lib/CMakeLists.txt

```diff
@@ -166,27 +162,42 @@ target_link_libraries(quda PUBLIC ${CUDA_cuda_driver_LIBRARY})
 
 # set up QUDA compile options
 target_compile_definitions(
+  quda PRIVATE $<$<CONFIG:DEVEL>:DEVEL> $<$<CONFIG:HOSTDEBUG>:HOST_DEBUG> $<$<CONFIG:DEVICEDEBUG>:DEVICE_DEBUG>
+               $<$<CONFIG:DEBUG>:HOST_DEBUGDEVICE_DEBUG> $<$<CONFIG:SANITIZE>:HOST_DEBUG>)
+
+target_compile_options(
   quda
-  PRIVATE
-  $<$<CONFIG:DEVEL>:DEVEL>
-  $<$<CONFIG:HOSTDEBUG>:HOST_DEBUG>
-  $<$<CONFIG:DEVICEDEBUG>:DEVICE_DEBUG>
-  $<$<CONFIG:DEBUG>:HOST_DEBUGDEVICE_DEBUG>
-  $<$<CONFIG:SANITIZE>:HOST_DEBUG>)
-
-target_compile_options(quda PRIVATE $<$<COMPILE_LANGUAGE:CUDA>: -ftz=true -prec-div=false -prec-sqrt=false>)
-target_compile_options(quda PRIVATE $<$<COMPILE_LANGUAGE:CUDA>: -Wno-deprecated-gpu-targets -arch=${QUDA_GPU_ARCH}
-                                    --expt-relaxed-constexpr>)
+  PRIVATE $<$<COMPILE_LANG_AND_ID:CUDA,NVIDIA>:
+          -ftz=true
+          -prec-div=false
+          -prec-sqrt=false>
+          $<$<COMPILE_LANG_AND_ID:CUDA,Clang>:
+          -fcuda-flush-denormals-to-zero
+          -fcuda-approx-transcendentals>)
+target_compile_options(
+  quda PRIVATE $<$<COMPILE_LANG_AND_ID:CUDA,Clang>:-Wno-unknown-cuda-version> $<$<COMPILE_LANG_AND_ID:CUDA,NVIDIA>:
+               -Wno-deprecated-gpu-targets -arch=${QUDA_GPU_ARCH} --expt-relaxed-constexpr>)
+
+target_compile_options(quda PRIVATE $<$<COMPILE_LANG_AND_ID:CUDA,NVIDIA>: -ftz=true -prec-div=false -prec-sqrt=false>)
+target_compile_options(quda PRIVATE $<$<COMPILE_LANG_AND_ID:CUDA,NVIDIA>: -Wno-deprecated-gpu-targets
+                                    -arch=${QUDA_GPU_ARCH} --expt-relaxed-constexpr>)
+target_compile_options(quda PRIVATE $<$<COMPILE_LANG_AND_ID:CUDA,Clang>: --cuda-path=${CUDAToolkit_TARGET_DIR}
```

Comment:
Question for my own understanding+sanity check: why is this `CUDA,Clang` and not, say, `CUDA,NVIDIA,Clang`?

Reply:
Added a question below, will mark this as resolved but I'm still curious as to my question.

Reply:
I believe the value returned by $<COMPILE_LANG_AND_ID> is  the pair [Lang],[CompilerID] so you can look for e.g.   CXX,Clang, or CXX,GNU (or whatever the GCC ID is) or CUDA,NVIDIA (that is if CUDA is the lang and NVCC is the compiler). It will not return a triplicate so this is not a. 'case in cond1|cond2|cond3` lie statement, but a does compile lang and ID match. If not, nothing is generated.  So we have separate CUDA,Clang one. Clang has different options from NVCC, e.g. no `--expt-relaxed-constexpr`